### PR TITLE
Document how to drop PHP target-dir attribute

### LINKED ISSUE #491: PHP: transitive dependency with target-dir not found
I'm trying to build a project that depends on `ml/json-ld`, but the build cannot find the library:

```
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires ml/json-ld 1.2.1 -> satisfiable by ml/json-ld[1.2.1].
    - ml/json-ld 1.2.1 requires ml/iri 1.1.4 -> could not be found in any version, there may be a>

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability se>
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
 - It's a private package and you forgot to add a custom repository to find it

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
```

I see the `composer.lock` specifies `ML/IRI` as `target-dir` for this package - not sure where that comes from.

I don't see that in the generated generic lock file. However, the generated derivation refers to `/nix/store/54fv7hp4dixphvfy3vr49g55j14nhrnb-ml-json-ld-1.2.1/lib/vendor/ml/json-ld/vendor/ml/iri` in the `repositoriesString`, which does not contain the library 'directly', but in a `ML/IRI` folder.

I have created a minimal example project showing the problem at https://github.com/nix-community/dream2nix/compare/main...raboof:dream2nix:php_transitive_with_target_dir

### PR DESCRIPTION
**Pull request by raboof**
Fixes #491 by documenting how to use flake `packageOverrides` to drop the attribute

### CODE REVIEW

File: tests/integration-d2n-flakes/tests/php_transitive_with_target_dir/flake.nix
Comment (DavHau) [inline tests/integration-d2n-flakes/tests/php_transitive_with_target_dir/flake.nix:1]: Please add a comment at the top of the file explaining what is tested and why

**Review by DavHau — COMMENTED** [review]
Please remove the flake.lock and composer.lock from the example

**Comment by raboof** [comment]
> Please remove the flake.lock and composer.lock from the example

composer.lock is actually needed as an input to the translator

**Comment by DavHau** [comment]
Thanks

**Pull request merged**